app/workers/account-generator.js

```diff
@@ -29,13 +56,22 @@ function makeId(len) {
 self.onmessage = event => {
     const {batchSize, studyId, extra, tag} = event.data;
     const records = [];
+    const idSet = [];
 
     for (let i = 0; i < batchSize; i++) {
         postMessage({
             progress: i
         });
 
-        const id = `${makeId(10)}${tag ? `-${tag}` : ''}`;
+        let id;
+
+        // Check for duplicated IDs. Unlikely, but possible.
+        while (!id || ~idSet.indexOf(id)) {
```

Comment:
For future reference, es6 provides `[].includes`, which is the same length, but more readable than the old `~indexOf` trick.